src/classes/RD2_VisualizeScheduleController.cls

```diff
@@ -80,6 +80,40 @@ public with sharing class RD2_VisualizeScheduleController {
         return upcomingDonations;
     }
 
+    /**
+     * @description Retrieves the current schedule of future donations to render in the UI
+     * @param recordId
+     * @return List<Schedule>
+     */
+    @AuraEnabled
+    public static List<Schedule> getSchedules(Id recordId) {
+
+        if (!RD2_EnablementService.isRecurringDonations2Enabled) {
```

Comment:
These first two validations (that throw an exception) are the same as in the `getInstallments()` method. I recommend abstracting them out into a separate method such as `validateRecurringDonationState()`.

Reply:
Let's extract common validation into a new method:
```
    /**
     * @description Validates if data can be displayed
     * @param recordId Recurring Donation Id
     * @return void
     */
    private static void validate(Id recordId) {
        if (!RD2_EnablementService.isRecurringDonations2Enabled) {
            throw new AuraHandledException(System.Label.RD2_ScheduleVisualizerErrorInvalidUsage);
        }

        if (recordId == null || recordId.getSObjectType() != npe03__Recurring_Donation__c.SObjectType) {
            throw new AuraHandledException(System.Label.RD2_ScheduleVisualizerErrorEnhancedRDNot);
        }   
    }
```

Can we rename `RD2_ScheduleVisualizerErrorEnhancedRDNot`? For example: `RD2_ScheduleVisualizerRequiresEnablement`?

Reply:
Can we rename this label?  It is already packaged isn't it?

Reply:
If this was previously merged into master then we shouldn't rename it.